salt/modules/reg.py

```diff
@@ -372,11 +378,14 @@ def read_value(hive, key, vname=None, use_32bit_registry=False):
             # RegQueryValueEx returns and accepts unicode data
```

Comment:
Conversion not required at all.....???

# RegQueryValueEx returns and accepts unicode data
Therefore only PY2 strings from arguments should be converted to Unicode.  As the result from RegQueryValueEx  should be Unicode.   I thought it was worked out that the win32reg returned mbcs and RegQueryValueEx return unicode

Reply:
It accepts and returns Unicode data... we only have to convert the return to mbcs to display properly on whatever encoding the system is using. Since it's not really supported that well in Py2, this seems to give the best results.

Reply:
> we only have to convert the return to mbcs to display properly on whatever encoding the system is using. 

It needs to return Unicode to the salt modules calling it PY2 and PY3, otherwise it will break win_pkg and other things.

When 2016.11 was released reg.py always to return Unicode, even it the parameters passed were PY2 strings.

The old mbcs function was used under win32reg to convert the result to Unicode, because the output from win32reg was mbcs and it needed conversion to Unicode.

You now have changed it to to convert Unicode to mbcs which is wrong.  !!!!!!!!!!

Reply:
Can you provide an example software package that is broken by this change so I can test this? I did quite a bit of testing on this to make sure it worked properly.